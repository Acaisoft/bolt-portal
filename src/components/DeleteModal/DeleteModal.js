import React from 'react'
import PropTypes from 'prop-types'
import {
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  DialogContentText,
  DialogTitle,
  withStyles,
} from '@material-ui/core'

import styles from './DeleteModal.styles'

export class DeleteModal extends React.Component {
  static propTypes = {
    classes: PropTypes.object.isRequired,
    open: PropTypes.bool.isRequired,
    handleClose: PropTypes.func.isRequired,
    handleSubmit: PropTypes.func.isRequired,
    name: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired,
  }

  render() {
    const { open, handleClose, handleSubmit, name, type } = this.props

    return (
      <div>
        <Dialog
          open={open}
          onClose={handleClose}
          aria-labelledby="delete-dialog-title"
          aria-describedby="delete-dialog-description"
        >
          <DialogTitle id="delete-dialog-title">Delete {type}</DialogTitle>
          <DialogContent>
            <DialogContentText id="delete-dialog-description">
              Are you sure want delete {name} {type}?
            </DialogContentText>
          </DialogContent>
          <DialogActions>
            <Button onClick={handleClose} color="primary">
              Cancel
            </Button>
            <Button
              onClick={handleSubmit}
              color="primary"
              variant="contained"
              autoFocus
            >
              Delete
            </Button>
          </DialogActions>
        </Dialog>
      </div>
    )
  }
}

export default withStyles(styles)(DeleteModal)
