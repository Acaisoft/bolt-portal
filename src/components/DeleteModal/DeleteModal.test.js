import React from 'react'
import { shallow } from 'enzyme'

import { DeleteModal } from './DeleteModal'

import { mocks } from '~utils/tests'

const initComponent = overrides => {
  const mockProps = {
    name: 'mockName',
    onClose: jest.fn(),
    onSubmit: jest.fn(),
    type: 'mockType',
  }
  const wrapper = shallow(
    <DeleteModal {...mockProps} {...overrides} classes={mocks.ClassesProxy} />
  )
  const instance = wrapper.instance()
  return { instance, mockProps, wrapper }
}

describe('component: DeleteModal', () => {
  describe('rendering', () => {
    it('should render without crashing', () => {
      const { wrapper } = initComponent()
      expect(wrapper).toBeTruthy()
    })
    it('should render as expected', () => {
      const { wrapper } = initComponent()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
