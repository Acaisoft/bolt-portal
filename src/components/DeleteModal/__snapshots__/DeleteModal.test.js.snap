// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: DeleteModal rendering should render as expected 1`] = `
<div>
  <WithStyles(Dialog)
    aria-describedby="delete-dialog-description"
    aria-labelledby="delete-dialog-title"
    onClose={[MockFunction]}
    open={false}
  >
    <WithStyles(DialogTitle)
      id="delete-dialog-title"
    >
      Delete 
      mockType
    </WithStyles(DialogTitle)>
    <WithStyles(DialogContent)>
      <WithStyles(DialogContentText)
        id="delete-dialog-description"
      >
        Are you sure want delete 
        mockName
         
        mockType
        ?
      </WithStyles(DialogContentText)>
    </WithStyles(DialogContent)>
    <WithStyles(DialogActions)>
      <WithStyles(Button)
        color="primary"
        onClick={[MockFunction]}
      >
        Cancel
      </WithStyles(Button)>
      <WithStyles(Button)
        autoFocus={true}
        color="primary"
        onClick={[MockFunction]}
        variant="contained"
      >
        Delete
      </WithStyles(Button)>
    </WithStyles(DialogActions)>
  </WithStyles(Dialog)>
</div>
`;
